share/functions/__fish_svn_prompt.fish

```diff
@@ -0,0 +1,169 @@
+set -g __fish_svn_prompt_char_token_stolen_char 'T'
+set -g __fish_svn_prompt_char_token_broken_char 'B'
+
+# ==============================
+
+# SVN status display variables
+# these are paired in groups of two:
+# 1. the character to display in the prompt
+# 2. the colour that should be used to display the character
+#
+# these variables are user-configurable and can be set to customize the display output
+
+set -g __fish_svn_prompt_char_added_display 'A'
+set -g __fish_svn_prompt_char_added_color green
+
+set -g __fish_svn_prompt_char_conflicted_display 'C'
+set -g __fish_svn_prompt_char_conflicted_color --underline magenta
+
+set -g __fish_svn_prompt_char_deleted_display 'D'
+set -g __fish_svn_prompt_char_deleted_color red
+
+set -g __fish_svn_prompt_char_ignored_display 'I'
+set -g __fish_svn_prompt_char_ignored_color --bold yellow
+
+set -g __fish_svn_prompt_char_modified_display 'M'
+set -g __fish_svn_prompt_char_modified_color blue
+
+set -g __fish_svn_prompt_char_replaced_display 'R'
+set -g __fish_svn_prompt_char_replaced_color cyan
+
+set -g __fish_svn_prompt_char_unversioned_external_display 'X'
+set -g __fish_svn_prompt_char_unversioned_external_color --underline cyan
+
+set -g __fish_svn_prompt_char_unversioned_display '?'
+set -g __fish_svn_prompt_char_unversioned_color purple
+
+set -g __fish_svn_prompt_char_missing_display '!'
+set -g __fish_svn_prompt_char_missing_color yellow
+
+set -g __fish_svn_prompt_char_versioned_obstructed_display '~'
+set -g __fish_svn_prompt_char_versioned_obstructed_color magenta
+
+set -g __fish_svn_prompt_char_locked_display 'L'
+set -g __fish_svn_prompt_char_locked_color --bold red
+
+set -g __fish_svn_prompt_char_scheduled_display '+'
+set -g __fish_svn_prompt_char_scheduled_color --bold green
+
+set -g __fish_svn_prompt_char_switched_display 'S'
+set -g __fish_svn_prompt_char_switched_color --bold blue
+
+set -g __fish_svn_prompt_char_token_present_display 'K'
+set -g __fish_svn_prompt_char_token_present_color --bold cyan
+
+set -g __fish_svn_prompt_char_token_other_display 'O'
+set -g __fish_svn_prompt_char_token_other_color --underline purple
+
+set -g __fish_svn_prompt_char_token_stolen_display 'T'
+set -g __fish_svn_prompt_char_token_stolen_color --bold purple
+
+set -g __fish_svn_prompt_char_token_broken_display 'B'
+set -g __fish_svn_prompt_char_token_broken_color --bold magenta
+
+
+# ==============================
+
+# this sets up an array of all the types of status codes that could be returned.
+set -g __fish_svn_prompt_flag_names added conflicted deleted ignored modified replaced unversioned_external unversioned missing locked scheduled switched token_present token_other token_stolen token_broken
+
+function __fish_svn_prompt_parse_status --argument flag_status_string --description "helper function that does pretty formatting on svn status"
+	# iterate over the different status types
+	for flag_index in (seq (count $__fish_svn_prompt_flag_names))
```

Comment:
Why not just iterate over __fish_svn_prompt_flag_names directly? The index is only used to access that.